### PR TITLE
BUILD-405: Install shared resource csi driver webhook alongside the s…

### PR DESCRIPTION
**Pull request by jkhelil**
…hared resource csi driver

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 
 	"github.com/openshift/csi-driver-shared-resource-operator/assets"
+	webhookdeployment "github.com/openshift/csi-driver-shared-resource-operator/pkg/deploymentcontroller"
```
Comment (coreydaley) [inline pkg/operator/starter.go:32]:
```suggestion
	"github.com/openshift/csi-driver-shared-resource-operator/pkg/deploymentcontroller"
```

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -147,9 +151,12 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		csidrivernodeservicecontroller.WithObservedProxyDaemonSetHook(),
 	)
 
-	if err != nil {
-		return err
-	}
+	webhookDeploymentController := webhookdeployment.NewWebHookDeploymentController(
```
Comment (coreydaley) [inline pkg/operator/starter.go:154]:
```suggestion
	webhookDeploymentController := deploymentcontroller.NewWebHookDeploymentController(
```

File: assets/assets.go [hunk 3]
```diff
@@ -4,10 +4,21 @@ import (
...
+// MustAsset reads and returns the content of the named file or panics
+// if something went wrong.
+func MustAsset(name string) []byte {
+	data, err := f.ReadFile(name)
+	if err != nil {
+		panic(err)
+	}
+
+	return data
+}
```
Comment (coreydaley) [inline assets/assets.go:24]:
```suggestion
// MustAsset reads and returns the content of the file at path or panics
// if something went wrong.
func MustAsset(path string) []byte {
	data, err := f.ReadFile(path)
	if err != nil {
		panic(err)
	}

	return data
}
```

**Review by coreydaley — CHANGES_REQUESTED** [review]

File: pkg/operator/starter.go (hunk 2 above)
Comment (jkhelil) [reply pkg/operator/starter.go]: fixed

File: pkg/operator/starter.go (hunk 1 above)
Comment (jkhelil) [reply pkg/operator/starter.go]: fixed

File: hack/deploy.sh [hunk 4]
```diff
@@ -16,20 +16,22 @@ rm -rf _deploy
 mkdir -p _deploy
 cp -r ../cluster-storage-operator/assets/csidriveroperators/shared-resource/* _deploy/
 
-operatorImage=${OPERATOR_IMAGE:-quay.io/openshift/origin-csi-driver-shared-resource-operator:latest}
+operatorImage=${OPERATOR_IMAGE:-quay.io/jkhelil/origin-csi-driver-shared-resource-operator:latest}
```
Comment (coreydaley) [inline hack/deploy.sh:19]: Does this need to be set back to `openshift`?

Comment (coreydaley) [reply hack/deploy.sh:19]: This is still not `openshift`?

**Comment by jkhelil** [comment]
we should fix 
```
1 pods found in best-effort QoS:
openshift-cluster-csi-drivers/shared-resource-csi-driver-webhook-5b5bf5746f-xthwv is running in best-effort QoS

```

**Pull request merged**